dbms/CMakeLists.txt

```diff
@@ -412,7 +412,19 @@ endif()
 
 if (USE_JEMALLOC)
     dbms_target_include_directories (SYSTEM BEFORE PRIVATE ${JEMALLOC_INCLUDE_DIR}) # used in Interpreters/AsynchronousMetrics.cpp
-    target_include_directories (clickhouse_common_io SYSTEM BEFORE PRIVATE ${JEMALLOC_INCLUDE_DIR}) # new_delete.cpp
+
+    macro (add_jemalloc target)
+        target_include_directories(${target} SYSTEM BEFORE PRIVATE ${JEMALLOC_INCLUDE_DIR})
+        target_link_libraries(${target} PRIVATE ${JEMALLOC_LIBRARIES})
+    endmacro()
```

Comment:
Why do we need to link jemalloc here?

Reply:
Because it uses jemalloc (comments below states from where), and with `--no-allow-shlib-undefined` it will not built

Reply:
Last time I tried linking libjemalloc.a into more than one binary and it crashed. Anyway after https://github.com/yandex/ClickHouse/pull/6602 and follow up https://github.com/yandex/ClickHouse/pull/6896 there is no need to link jemalloc here. Actually it should not be linked.

Reply:
>Last time I tried linking libjemalloc.a into more than one binary and it crashed

Good point.

>Anyway after #6602

Indeed, because it removes `-Wl,--no-undefined` from `CMAKE_SHARED_LINKER_FLAGS`, I will remove this patch

Reply:
9152a832c47bca3c2663443ed7b8239fb24b59a9 (P.S. it has a typo I guess `s/target_link_libraries/target_link_options/`, although this does not change anything) breaks this again:

```
cmake -DCMAKE_BUILD_TYPE=release -DUSE_STATIC_LIBRARIES=OFF -DCLICKHOUSE_SPLIT_BINARY=ON -DSPLIT_SHARED_LIBRARIES=ON -DUNBUNDLED=ON

ninja dbms/libclickhouse_common_io.so
/usr/bin/ld: dbms/CMakeFiles/clickhouse_common_io.dir/src/Common/new_delete.cpp.o: in function `operator delete(void*, unsigned long)':
new_delete.cpp:(.text+0xa8): undefined reference to `sdallocx'

```

*(same for `-DCMAKE_BUILD_TYPE=debug`)*`

Reply:
target_link_options is only available for cmake 3.13

Reply:
and yes, I didn't test unbundled shared build because CI doesn't  either

Reply:
>target_link_options is only available for cmake 3.13

got it

>and yes, I didn't test unbundled shared build because CI doesn't either

so maybe smth like above? also maybe it is a good idea to add unbundled build to CI?

```patch
diff --git a/dbms/CMakeLists.txt b/dbms/CMakeLists.txt
index 2c8f99b973..0957ddb76a 100644
--- a/dbms/CMakeLists.txt
+++ b/dbms/CMakeLists.txt
@@ -414,6 +414,11 @@ endif()
 if (USE_JEMALLOC)
     dbms_target_include_directories (SYSTEM BEFORE PRIVATE ${JEMALLOC_INCLUDE_DIR}) # used in Interpreters/AsynchronousMetrics.cpp
     target_include_directories (clickhouse_common_io SYSTEM BEFORE PRIVATE ${JEMALLOC_INCLUDE_DIR}) # new_delete.cpp
+    # common/memory.h
+    if (MAKE_STATIC_LIBRARIES OR NOT SPLIT_SHARED_LIBRARIES)
+    else()
+        target_link_libraries(clickhouse_common_io PRIVATE ${JEMALLOC_LIBRARIES})
+    endif()
 endif ()

 dbms_target_include_directories (PUBLIC ${DBMS_INCLUDE_DIR} PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/src/Formats/include)

```